domain-models-interface-extensions/src/main/java/org/folio/rest/tools/PomReader.java

```diff
@@ -34,81 +38,102 @@
   @SuppressWarnings("checkstyle:methodlength")
   private PomReader() {
     try {
-      System.out.print("Attempting to read in the module name from....");
-      MavenXpp3Reader mavenreader = new MavenXpp3Reader();
-      Model model = null;
       String currentRunningJar =
           PomReader.class.getProtectionDomain().getCodeSource().getLocation().toURI().getPath();
-      System.out.println(currentRunningJar);
-      if(currentRunningJar != null && (currentRunningJar.contains("domain-models-runtime")
-          || currentRunningJar.contains("domain-models-interface-extensions"))){
-        //the runtime is the jar run when deploying during unit tests
-        //the interface-extensions is the jar run when running build time tools,
-        //like MDGenerator, ClientGenerator, etc..
-        //this is build time - not runtime, so just use the pom
-        File pomFile = new File("pom.xml");
-        model = mavenreader.read(new FileReader(pomFile));
-      }
-      else{
-        //this is runtime, the jar called via java -jar is the module's jar
-        ClassPath classPath = ClassPath.from(Thread.currentThread().getContextClassLoader());
-        Set<ClassPath.ResourceInfo> resources = classPath.getResources();
-        for (ClassPath.ResourceInfo info : resources) {
-          if(info.getResourceName().endsWith("pom.xml")){
-            //maven sets the classpath order according to the pom, so the poms project will be the first entry
-            model = mavenreader.read(PomReader.class.getResourceAsStream("/"+info.getResourceName()));
-            break;
-          }
-        }
-      }
-      if(model.getParent() != null){
-        moduleName = model.getParent().getArtifactId();
-        version = model.getParent().getVersion();
-      }
-      else{
-        moduleName = model.getArtifactId();
-      }
+      boolean readCurrent = currentRunningJar != null && (currentRunningJar.contains("domain-models-runtime")
+          || currentRunningJar.contains("domain-models-interface-extensions"));
+      readIt(readCurrent);
+    } catch (Exception e) {
+      log.error(e.getMessage(), e);
```

Comment:
The exception should be rethrown as a RuntimeException to fail the deployment if we cannot read the module name.

Reply:
Yes. Agreed. Hesitated because that will change behavior.

Reply:
For now, re-throw a Runtime exception. I don't know how to otherwise do it with this enum. If you think it's a big deal I think we should make a follow up issue. I'd like to get fix out.